src/basecamp.cpp

```diff
@@ -704,21 +704,18 @@ void basecamp::form_storage_zones( map &here, const tripoint_abs_ms &abspos )
                 60, get_owner() );
         // Find the nearest unsorted zone to dump objects at
         if( !zones.empty() ) {
-            if( zones != storage_zones ) {
-                std::unordered_set<tripoint_abs_ms> src_set;
-                for( const zone_data *zone : zones ) {
-                    for( const tripoint_abs_ms &p : tripoint_range<tripoint_abs_ms>(
-                             zone->get_start_point(), zone->get_end_point() ) ) {
-                        src_set.emplace( p );
-                    }
+            std::unordered_set<tripoint_abs_ms> src_set;
+            for( const zone_data *zone : zones ) {
+                for( const tripoint_abs_ms &p : tripoint_range<tripoint_abs_ms>(
+                         zone->get_start_point(), zone->get_end_point() ) ) {
+                    src_set.emplace( p );
                 }
-                set_storage_tiles( src_set );
             }
+            set_storage_tiles( src_set );
```

Comment:
Note that `src_set` here is the new local set of points, while `set_storage_tiles` is going to set the `src_set` field on the basecamp object. The same name for the equivalent variables, with a different name for the setter function, confused me for a bit.